drivers/watchdog/wdt_mcux_imx_rtwdog.c

```diff
@@ -0,0 +1,222 @@
+
+	timeoutCycle = MSEC_TO_RTWDOG_TICKS(clock_freq, div, cfg->window.max);
+	if (timeoutCycle > U16_MAX) {
+		LOG_ERR("Invalid window max");
+		return -EINVAL;
+	}
+	data->wdog_config.timeoutValue = timeoutCycle;
+
+	if (cfg->window.min) {
+		data->wdog_config.enableWindowMode = true;
+		data->wdog_config.windowValue =
+			MSEC_TO_RTWDOG_TICKS(clock_freq, div, cfg->window.min);
+	} else {
+		data->wdog_config.enableWindowMode = false;
+		data->wdog_config.windowValue = 0;
+	}
+
+	data->wdog_config.enableInterrupt = cfg->callback != NULL;
+	data->callback = cfg->callback;
+	data->timeout_valid = true;
+
+	return 0;
+}
+
+static int mcux_rtwdog_feed(const struct device *dev, int channel_id)
+{
+	const struct mcux_rtwdog_config *config = dev->config;
+	RTWDOG_Type *base = config->base;
+
+	if (channel_id != 0) {
+		LOG_ERR("Invalid channel id");
+		return -EINVAL;
+	}
+
+	RTWDOG_Refresh(base);
+	LOG_DBG("Fed the watchdog");
+
+	return 0;
+}
+
+static void mcux_rtwdog_isr(const struct device *dev)
+{
+	const struct mcux_rtwdog_config *config = dev->config;
+	struct mcux_rtwdog_data *data = dev->data;
+	RTWDOG_Type *base = config->base;
+	uint32_t flags;
+
+	flags = RTWDOG_GetStatusFlags(base);
+	RTWDOG_ClearStatusFlags(base, flags);
+
+	if (data->callback) {
+		data->callback(dev, 0);
+	}
+}
+
+static int mcux_rtwdog_init(const struct device *dev)
+{
+	const struct mcux_rtwdog_config *config = dev->config;
+	int ret;
+
+	config->irq_config_func(dev);
+
+	ret = pinctrl_apply_state(config->pcfg, PINCTRL_STATE_DEFAULT);
+	if (ret < 0 && ret != -ENOENT) {
+		return ret;
+	}
+
+	return 0;
+}
+
+static DEVICE_API(wdt, mcux_rtwdog_api) = {
+	.setup = mcux_rtwdog_setup,
+	.disable = mcux_rtwdog_disable,
+	.install_timeout = mcux_rtwdog_install_timeout,
+	.feed = mcux_rtwdog_feed,
+};
+
+static void mcux_rtwdog_config_func(const struct device *dev);
+
+PINCTRL_DT_INST_DEFINE(0);
+
+#define TO_RTWDOG_CLK_SRC(val) _DO_CONCAT(kRTWDOG_ClockSource, val)
+#define TO_RTWDOG_CLK_DIV(val) _DO_CONCAT(kRTWDOG_ClockPrescalerDivide, val)
+
+static const struct mcux_rtwdog_config mcux_rtwdog_config = {
+	.base = (RTWDOG_Type *)DT_INST_REG_ADDR(0),
+	.irq_config_func = mcux_rtwdog_config_func,
+	.pcfg = PINCTRL_DT_INST_DEV_CONFIG_GET(0),
+#if DT_NODE_HAS_PROP(DT_INST_PHANDLE(0, clocks), clock_frequency)
+	.clock_frequency = DT_INST_PROP_BY_PHANDLE(0, clocks, clock_frequency),
+#else  /* !DT_NODE_HAS_PROP(DT_INST_PHANDLE(0, clocks), clock_frequency) */
+	.clock_dev = DEVICE_DT_GET(DT_INST_CLOCKS_CTLR(0)),
+	.clock_subsys = (clock_control_subsys_t)DT_INST_CLOCKS_CELL(0, name),
+#endif /* DT_NODE_HAS_PROP(DT_INST_PHANDLE(0, clocks), clock_frequency) */
+	.clk_source = TO_RTWDOG_CLK_SRC(DT_INST_PROP(0, clk_source)),
+	.clk_divider = TO_RTWDOG_CLK_DIV(DT_INST_PROP(0, clk_divider)),
+};
+
+static struct mcux_rtwdog_data mcux_rtwdog_data;
+
+DEVICE_DT_INST_DEFINE(0, &mcux_rtwdog_init, NULL, &mcux_rtwdog_data, &mcux_rtwdog_config,
```

Comment:
Let's not use this syntax here, this driver can easily be multi instance. Take a look at this driver for an example of a multi-instance watchdog setup: https://github.com/zephyrproject-rtos/zephyr/blob/537b52e7fe8f183fc7e51f5d0948464cf71ffa38/drivers/watchdog/wdt_gecko.c#L281